<h1> 1.INTRODCUTION </h1>

<h2>1.a. Discussion of the Background </h2>

<h4> Silicon Valley is the heart of global innovation, technology and social media. It is home to big tech giants like Apple, Facebook, Google, Microsoft, Linkedin, Uber and many more. With such big names come tremondous job opportunities which pulls people all over the world to migrate and make this silicon valley their home. But this huge influx of people eventually causes its own problems like, high property rates, cost of living and crazy traffic. While this fast paced life has its allure, it does not suite everyone and some prefer to have the advantages of silicon valley balanced with a somewhat quite and relaxing life more closer to the nature. </h4>
    
<h4>For these individuals the "SILICON FOREST" is a godsent! The silicon forest is a group of tech companies eastablished in the Willamett Valley in Oregon. Sice 1980's to date the area around silicon forest has seen a rise in population, infrastructure growth and development, expansion of cities, growth in housing and night life. It has almost all the pros of silicon valley with a more chilled out lifestyle. The Silicon Forest has drawn a crowd similar to silicon valley to its midst. </h4>
 

<h2> 1.b. Description of Problem </h2>

<h4> Let us consider an individual in intersted in starting a Restaurant with a unique idea which has not been 
     tested before. They are not ceratin if it will be embraced by the masses or be a big failure. They want to
    try running a successfull business at a smaller scale, to test the waters before investing huge amounts and opening at a large 
    scale in heavily populous cities like San Francisco and San Jose. They hear about the silicon forest and feel as the client base might be similar it would be a good idea to consider opening a business there. They want to explore Oregon to start a busniess. </h4>
 <h4> So in this Capstone project I would like to address this problem. We will try to find out which county in Oregon has the most venues so as to select that county as the place of busniess. To explore the neighborhoods in that county, pitting them each other in the battle of neighborhoods.</h4>
 
 <h4> This kind of analysis will help not only the people seeking to open a business in popular places but with lower property rates, it will also help people who are seeking otions of places similar to silicon valley to relocate for a quiter life. </h4>

<h1> 2.DESCRIPTION OF DATA </h1>

<h2> 2.a. For the list of Counties </h2>
<h4> Data Source: https://en.wikipedia.org/wiki/List_of_counties_in_Oregon </h4>
   <h4>  Data Description: Will scrape the wikipidea page with the table contatining the information of all the counties of Oregon. </h4>
   
 <h2>   2.b. For Latittude and Longitutdes</h2>
   <h4>  Data Source: Geopy</h4>
   <h4> Data Description: The wikipedia page does not contain the latitudes and longitude information. So the data was obtained by using the Geocoder class of the Geopy client. </h4>
   
   <h2> 2.c. List of Venues and Venue categories.</h2>
   <h4> Data Source: Foursquare API's</h4>
 <h4>   Data Description: By using the Foursquare API we can get the information of the venues in the counties. Based on this this data we can further analyse the data by selecting the county with the highest number of venues. We can again use Foursquare API to get data about the venue categories in the neighborhoods of the selected county. </h4>

<h1> 3.METHODOLOGY </h1>
<h2> 3.a Data Preparation </h2>
<h4> As the State of Oregon is the state of interest, I first scraped the wikipidea page for information about counties in Oregon. I used pandas to scrape the table on wikipedia page and convert it to a pandas data frame. </h4>

In [1]:
import pandas as pd
import numpy as np

!conda install -c conda-forge lxml --yes

!conda install -c conda-forge geopy --yes 
from geopy.geocoders import Nominatim


Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.8.3
  latest version: 4.8.4

Please update conda by running

    $ conda update -n base -c defaults conda



## Package Plan ##

  environment location: /home/jupyterlab/conda/envs/python

  added / updated specs:
    - lxml


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    libxml2-2.9.10             |       he19cac6_1         1.2 MB
    libxslt-1.1.34             |       hc22bd24_0         432 KB
    lxml-4.5.2                 |   py36h17c4326_0         1.3 MB  conda-forge
    openssl-1.1.1g             |       h516909a_1         2.1 MB  conda-forge
    ------------------------------------------------------------
                                           Total:         5.1 MB

The following NEW packages will be INSTALLED:

  libxslt            pkgs/main/linux-64::libxslt-1.1.34-hc22bd24_0

In [93]:
df_pdx = pd.read_html('https://en.wikipedia.org/wiki/List_of_counties_in_Oregon')[1]

In [94]:
df_pdx.head()

County  FIPS code[3] County seat[4]  Est.[4]  \
0      Baker County             1     Baker City     1862   
1     Benton County             3      Corvallis     1847   
2  Clackamas County             5    Oregon City     1843   
3    Clatsop County             7        Astoria     1844   
4   Columbia County             9   Saint Helens     1854   

                                           Origin[5]  \
0      Wasco County, Union County and Malheur County   
1                                        Polk County   
2  One of the original four districts of the Oreg...   
3  Northern and western portions of the original ...   
4                 Northern half of Washington County   

                                        Etymology[5]  Population[6]  \
0  Edward Dickinson Baker, a senator from Oregon ...          16510   
1  Thomas Hart Benton (1782–1858), a U.S. senator...          91320   
2  The Clackamas Indians, a Native American tribe...         404980   
3  Clatsop tribe of Native Americans, who lived a...          38225   
4  Columbia River, which forms the county's north...          50795   

                  Area[4]  Map  
0  3,068 sq mi(7,946 km2)  NaN  
1    676 sq mi(1,751 km2)  NaN  
2  1,868 sq mi(4,838 km2)  NaN  
3    827 sq mi(2,142 km2)  NaN  
4    657 sq mi(1,702 km2)  NaN

<h4> I further cleaned the dataframe by dropping columns which we will not be using for analysis. </h4>

In [95]:
df_pdx=df_pdx.drop(['Map','Origin[5]','FIPS code[3]','Etymology[5]','Area[4]'], axis = 1) 
df_pdx.head()

County County seat[4]  Est.[4]  Population[6]
0      Baker County     Baker City     1862          16510
1     Benton County      Corvallis     1847          91320
2  Clackamas County    Oregon City     1843         404980
3    Clatsop County        Astoria     1844          38225
4   Columbia County   Saint Helens     1854          50795

<h4> For using the Foursquare API to get information of the venues and restaurants in the counties, 
the latitude and longitude information is required. This information is missing in our dataframe. 
So I used the geocoder class of Geopy to get the coordinates of all the counties of Oregon. </h4>

In [5]:
geolocator = Nominatim(user_agent="Oregon_explorer")

df_pdx['Area[4]'] = df_pdx['County'].apply(lambda x: x + ", Oregon, USA").apply(geolocator.geocode).apply(lambda x: (x.latitude, x.longitude))
df_pdx[['Latitude','Longitude']] = df_pdx['Area[4]'].apply(pd.Series)
df_pdx.drop(['Area[4]'],axis=1,inplace=True)
df_pdx

County County seat[4]  Est.[4]   Latitude   Longitude
0        Baker County     Baker City     1862  44.725964 -117.620482
1       Benton County      Corvallis     1847  44.494937 -123.406568
2    Clackamas County    Oregon City     1843  45.160882 -122.230504
3      Clatsop County        Astoria     1844  45.980728 -123.668750
4     Columbia County   Saint Helens     1854  45.928020 -123.082293
5         Coos County       Coquille     1853  43.218414 -124.109621
6        Crook County     Prineville     1882  44.146029 -120.383948
7        Curry County     Gold Beach     1855  42.426543 -124.216879
8    Deschutes County           Bend     1916  43.819484 -121.166966
9      Douglas County       Roseburg     1852  43.192211 -123.113596
10     Gilliam County         Condon     1885  45.329874 -120.220220
11       Grant County    Canyon City     1864  44.464824 -119.056528
12      Harney County          Burns     1889  43.055195 -119.024026
13  Hood River County     Hood River     1908  45.531164 -121.647559
14     Jackson County        Medford     1852  42.398015 -122.754095
15   Jefferson County         Madras     1914  44.607250 -121.246314
16   Josephine County    Grants Pass     1856  42.355437 -123.560271
17     Klamath County  Klamath Falls     1882  42.605257 -121.739544
18        Lake County       Lakeview     1874  42.770285 -120.452833
19        Lane County         Eugene     1851  43.931457 -122.857585
20     Lincoln County        Newport     1893  44.620174 -123.927691
21        Linn County         Albany     1847  44.496890 -122.548598
22     Malheur County           Vale     1887  43.102348 -117.691689
23      Marion County          Salem     1843  44.896442 -122.720325
24      Morrow County        Heppner     1885  45.360901 -119.605246
25   Multnomah County       Portland     1854  45.509380 -122.400641
26        Polk County         Dallas     1845  44.879673 -123.459959
27     Sherman County           Moro     1889  45.394427 -120.696375
28   Tillamook County      Tillamook     1853  45.460293 -123.726679
29    Umatilla County      Pendleton     1862  45.641700 -118.767112
30       Union County      La Grande     1864  45.268001 -118.025747
31     Wallowa County     Enterprise     1887  45.558261 -117.199017
32       Wasco County     The Dalles     1854  45.043023 -121.215421
33  Washington County      Hillsboro     1843  45.560188 -123.058791
34     Wheeler County         Fossil     1899  44.676838 -120.048438
35     Yamhill County    McMinnville     1843  45.229840 -123.218041

<h2> 3.b. Data Analysis </h2>
<h4> Now we have all the information to call the Foursquare API.
Lets now define the url and call the function to get the venues present in these counties of Oregon. </h4>

In [6]:
CLIENT_ID = 'D502JG0ELHOWV5WZ2ECT32GRHHNS4SLYKPMRRV52G0EFNQNJ' #  Foursquare ID
CLIENT_SECRET = 'ECYPGKMQ3C400H2FC3T2I40JBYSWH3DXSAVNCZQ04ZQLXJPL' #  Foursquare Secret
VERSION = '20180604'
LIMIT = 500
print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: D502JG0ELHOWV5WZ2ECT32GRHHNS4SLYKPMRRV52G0EFNQNJ
CLIENT_SECRET:ECYPGKMQ3C400H2FC3T2I40JBYSWH3DXSAVNCZQ04ZQLXJPL


In [7]:
import requests 
from pandas.io.json import json_normalize

In [8]:
def getNearbyVenues(names, latitudes, longitudes, radius=5000):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        print(name + ': ' + str(len(results)))
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['County', 
                  'County Latitude', 
                  'County Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    return(nearby_venues)

In [9]:
pdx_venues = getNearbyVenues(names=df_pdx['County'],
                                   latitudes=df_pdx['Latitude'],
                                   longitudes=df_pdx['Longitude']
                                  )

pdx_venues.shape

Baker County: 0
Benton County: 5
Clackamas County: 2
Clatsop County: 4
Columbia County: 0
Coos County: 2
Crook County: 0
Curry County: 0
Deschutes County: 0
Douglas County: 1
Gilliam County: 0
Grant County: 0
Harney County: 0
Hood River County: 9
Jackson County: 6
Jefferson County: 6
Josephine County: 5
Klamath County: 0
Lake County: 0
Lane County: 10
Lincoln County: 9
Linn County: 2
Malheur County: 0
Marion County: 4
Morrow County: 3
Multnomah County: 100
Polk County: 4
Sherman County: 1
Tillamook County: 4
Umatilla County: 43
Union County: 5
Wallowa County: 0
Wasco County: 0
Washington County: 38
Wheeler County: 0
Yamhill County: 46


(309, 7)

<h4> From the above counts of venue in each county it is clear that maximum number of
venues are in the MULTNOMAH COUNTY. So let us carry out further analysis on the Multnomah County </h4>
<h4> I found a list online of all neighborhoods in the multnomah county. I am going to start the analysis by converting the list to a dataframe and then using Geopy to get coordinates and Foursquare API to get the venue information. </h4>

In [10]:
multnomah_neighborhoods = ['Alameda', 'Albina', 'Arleta', 'Barnes Heights', 'Bonneville', 'Bonny Slope', 'Bridal Veil', 'Bridlemile', 'Brooklyn',
                           'Burlingame', 'Burlington', 'Capitol Hill', 'Carson Heights', 'Corbett', 'Dodson', 'Dunthorpe', 'East Portland',
                        'Eastmoreland', 'Englewood', 'Errol Heights', 'Faloma', 'Folkenberg', 'Gilbert', 'Glencullen', 'Green Hills', 'Hazelwood',
                        'Hillsdale', 'Holbrook', 'Hollywood', 'Interlachen', 'Irvington', 'Johns Landing', 'Kellogg Park', 'Kenton', 'Kings Heights', 
                           'Latourell', 'Laurelhurst', 'Lents', 'Linnton', 'Maplewood', 'Marquam Hill', 'Montavilla', 'Multnomah',
                        'North Portland', 'Orient', 'Parkrose', 'Piedmont', 'Pleasant Home', 'Portland Heights', 'Powell Valley', 'Powellhurst',
                        'Riverdale', 'Riverwood', 'Rockwood', 'Russellville', 'Saint Johns', 'Sellwood', 'Springdale', 'Sylvan', 'Twelvemile Corner',
                       'University Park', 'Vermont Hills', 'Villa Ridge', 'Warrendale', 'West Portland', 'West Portland Park', 'Westmoreland',
                        'Whitwood Court', 'Willamette Heights', 'Woodstock']


In [11]:
df_mult = pd.DataFrame(multnomah_neighborhoods, columns=['Neighborhoods'])
df_mult.head()

Neighborhoods
0         Alameda
1          Albina
2          Arleta
3  Barnes Heights
4      Bonneville

In [12]:
geolocator = Nominatim(user_agent="Multnomah_explorer")

df_mult['Area[4]'] = df_mult['Neighborhoods'].apply(lambda x: x + ", Oregon, USA").apply(geolocator.geocode).apply(lambda x: (x.latitude, x.longitude))
df_mult[['Latitude','Longitude']] = df_mult['Area[4]'].apply(pd.Series)
df_mult.drop(['Area[4]'],axis=1,inplace=True)
df_mult

Neighborhoods   Latitude   Longitude
0              Alameda  45.548631 -122.636481
1               Albina  45.546631 -122.667635
2               Arleta  45.486459 -122.589298
3       Barnes Heights  45.532289 -122.753756
4           Bonneville  45.644265 -121.940787
..                 ...        ...         ...
65  West Portland Park  45.447282 -122.720814
66        Westmoreland  45.476230 -122.647039
67      Whitwood Court  45.585395 -122.772044
68  Willamette Heights  45.535118 -122.717597
69           Woodstock  45.480705 -122.614549

[70 rows x 3 columns]

In [13]:
mult_venues = getNearbyVenues(names=df_mult['Neighborhoods'],
                                   latitudes=df_mult['Latitude'],
                                   longitudes=df_mult['Longitude']
                                  )

mult_venues.shape

Alameda: 100
Albina: 100
Arleta: 100
Barnes Heights: 100
Bonneville: 45
Bonny Slope: 44
Bridal Veil: 24
Bridlemile: 100
Brooklyn: 100
Burlingame: 100
Burlington: 16
Capitol Hill: 87
Carson Heights: 100
Corbett: 18
Dodson: 16
Dunthorpe: 100
East Portland: 100
Eastmoreland: 100
Englewood: 57
Errol Heights: 100
Faloma: 100
Folkenberg: 13
Gilbert: 100
Glencullen: 100
Green Hills: 100
Hazelwood: 100
Hillsdale: 100
Holbrook: 13
Hollywood: 100
Interlachen: 100
Irvington: 100
Johns Landing: 100
Kellogg Park: 100
Kenton: 100
Kings Heights: 100
Latourell: 24
Laurelhurst: 100
Lents: 100
Linnton: 89
Maplewood: 100
Marquam Hill: 100
Montavilla: 100
Multnomah: 100
North Portland: 100
Orient: 16
Parkrose: 100
Piedmont: 100
Pleasant Home: 14
Portland Heights: 100
Powell Valley: 55
Powellhurst: 100
Riverdale: 100
Riverwood: 100
Rockwood: 100
Russellville: 100
Saint Johns: 100
Sellwood: 100
Springdale: 27
Sylvan: 100
Twelvemile Corner: 100
University Park: 100
Vermont Hills: 100
Villa Ridge: 100
Warrend

(5862, 7)

In [14]:
print ('{} Unique Categories in Multnomah County'.format(mult_venues['Venue Category'].value_counts().shape[0]))

262 Unique Categories in Multnomah County


<h4> The Foursquare returned 262 unique categories. We are interested only in restaurants.
So let us clean our dataframe and consider only those venues which are Restaurants. </h4>

In [15]:
df_multresto = mult_venues[mult_venues['Venue Category'].astype(str).str.contains('Restaurant')]

df_multresto.head()

County  County Latitude  County Longitude                Venue  \
1   Alameda        45.548631       -122.636481  Pine State Biscuits   
2   Alameda        45.548631       -122.636481           Kargi Gogo   
4   Alameda        45.548631       -122.636481    Bollywood Theater   
8   Alameda        45.548631       -122.636481             Urdaneta   
15  Alameda        45.548631       -122.636481                Beast   

    Venue Latitude  Venue Longitude                   Venue Category  
1        45.558981      -122.642697  Southern / Soul Food Restaurant  
2        45.559113      -122.634038              Dumpling Restaurant  
4        45.559270      -122.644011                Indian Restaurant  
8        45.559136      -122.634119                 Tapas Restaurant  
15       45.562462      -122.634985                French Restaurant

In [16]:
print (np.unique(df_multresto['Venue Category']))

['American Restaurant' 'Argentinian Restaurant' 'Asian Restaurant'
 'Belgian Restaurant' 'Brazilian Restaurant' 'Cajun / Creole Restaurant'
 'Cambodian Restaurant' 'Chinese Restaurant' 'Cuban Restaurant'
 'Dim Sum Restaurant' 'Dumpling Restaurant' 'Eastern European Restaurant'
 'Falafel Restaurant' 'Fast Food Restaurant' 'French Restaurant'
 'German Restaurant' 'Greek Restaurant' 'Hawaiian Restaurant'
 'Indian Restaurant' 'Indonesian Restaurant' 'Israeli Restaurant'
 'Italian Restaurant' 'Japanese Restaurant' 'Korean Restaurant'
 'Latin American Restaurant' 'Mediterranean Restaurant'
 'Mexican Restaurant' 'Middle Eastern Restaurant'
 'Modern European Restaurant' 'Moroccan Restaurant'
 'New American Restaurant' 'Peruvian Restaurant' 'Ramen Restaurant'
 'Restaurant' 'Russian Restaurant' 'Scandinavian Restaurant'
 'Seafood Restaurant' 'Southern / Soul Food Restaurant' 'Sushi Restaurant'
 'Tapas Restaurant' 'Thai Restaurant' 'Vegetarian / Vegan Restaurant'
 'Vietnamese Restaurant']


In [17]:
print ('{} Unique Restaurant Categories in Multnomah County'.format(df_multresto['Venue Category'].value_counts().shape[0]))

43 Unique Restaurant Categories in Multnomah County


In [18]:
print (df_multresto['Venue Category'].value_counts()[0:10])

Mexican Restaurant               148
Thai Restaurant                   96
American Restaurant               95
Vietnamese Restaurant             84
Italian Restaurant                75
Seafood Restaurant                60
Sushi Restaurant                  46
Vegetarian / Vegan Restaurant     44
Restaurant                        39
Chinese Restaurant                34
Name: Venue Category, dtype: int64


In [26]:
df_multresto.rename(columns = {'County':'Neighborhoods','County Latitude':'Neighborhood Latitude','County Longitude':'Neighborhood Longitude'}, inplace = True) 
df_multresto.head().reset_index()
#df_multresto=df_multresto.drop(['index'], axis = 1)

index Neighborhoods  Neighborhood Latitude  Neighborhood Longitude  \
0      1       Alameda              45.548631             -122.636481   
1      2       Alameda              45.548631             -122.636481   
2      4       Alameda              45.548631             -122.636481   
3      8       Alameda              45.548631             -122.636481   
4     15       Alameda              45.548631             -122.636481   

                 Venue  Venue Latitude  Venue Longitude  \
0  Pine State Biscuits       45.558981      -122.642697   
1           Kargi Gogo       45.559113      -122.634038   
2    Bollywood Theater       45.559270      -122.644011   
3             Urdaneta       45.559136      -122.634119   
4                Beast       45.562462      -122.634985   

                    Venue Category  
0  Southern / Soul Food Restaurant  
1              Dumpling Restaurant  
2                Indian Restaurant  
3                 Tapas Restaurant  
4                French Restaurant

<h4> Let us now analyse each neighborhood to get TOP 5 most common venues.
Lets do this by first creating a dataframe with one hot encoding. </h4>
<h4> Then by using groupy on the neighborhood column we will calculate the frequency of occurance for each of the 
unique venues </h4>

In [27]:
pdx_onehot = pd.get_dummies(df_multresto[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
pdx_onehot['Neighborhoods'] = df_multresto['Neighborhoods'] 
# move neighborhood column to the first column
fixed_columns = [pdx_onehot.columns[-1]] + list(pdx_onehot.columns[:-1])
pdx_onehot = pdx_onehot[fixed_columns]

pdx_onehot.head()

Neighborhoods  American Restaurant  Argentinian Restaurant  \
1        Alameda                    0                       0   
2        Alameda                    0                       0   
4        Alameda                    0                       0   
8        Alameda                    0                       0   
15       Alameda                    0                       0   

    Asian Restaurant  Belgian Restaurant  Brazilian Restaurant  \
1                  0                   0                     0   
2                  0                   0                     0   
4                  0                   0                     0   
8                  0                   0                     0   
15                 0                   0                     0   

    Cajun / Creole Restaurant  Cambodian Restaurant  Chinese Restaurant  \
1                           0                     0                   0   
2                           0                     0                   0   
4                           0                     0                   0   
8                           0                     0                   0   
15                          0                     0                   0   

    Cuban Restaurant  ...  Restaurant  Russian Restaurant  \
1                  0  ...           0                   0   
2                  0  ...           0                   0   
4                  0  ...           0                   0   
8                  0  ...           0                   0   
15                 0  ...           0                   0   

    Scandinavian Restaurant  Seafood Restaurant  \
1                         0                   0   
2                         0                   0   
4                         0                   0   
8                         0                   0   
15                        0                   0   

    Southern / Soul Food Restaurant  Sushi Restaurant  Tapas Restaurant  \
1                                 1                 0                 0   
2                                 0                 0                 0   
4                                 0                 0                 0   
8                                 0                 0                 1   
15                                0                 0                 0   

    Thai Restaurant  Vegetarian / Vegan Restaurant  Vietnamese Restaurant  
1                 0                              0                      0  
2                 0                              0                      0  
4                 0                              0                      0  
8                 0                              0                      0  
15                0                              0                      0  

[5 rows x 44 columns]

In [28]:
pdx_grouped = pdx_onehot.groupby('Neighborhoods').mean().reset_index()
print(pdx_grouped.shape)


pdx_grouped.head()

(61, 44)


Neighborhoods  American Restaurant  Argentinian Restaurant  \
0         Alameda             0.034483                0.034483   
1          Albina             0.041667                0.041667   
2          Arleta             0.086957                0.000000   
3  Barnes Heights             0.058824                0.000000   
4      Bonneville             0.500000                0.000000   

   Asian Restaurant  Belgian Restaurant  Brazilian Restaurant  \
0          0.000000                 0.0                   0.0   
1          0.000000                 0.0                   0.0   
2          0.043478                 0.0                   0.0   
3          0.000000                 0.0                   0.0   
4          0.000000                 0.0                   0.0   

   Cajun / Creole Restaurant  Cambodian Restaurant  Chinese Restaurant  \
0                   0.000000                   0.0            0.000000   
1                   0.000000                   0.0            0.000000   
2                   0.043478                   0.0            0.086957   
3                   0.000000                   0.0            0.000000   
4                   0.000000                   0.0            0.000000   

   Cuban Restaurant  ...  Restaurant  Russian Restaurant  \
0               0.0  ...    0.000000                 0.0   
1               0.0  ...    0.000000                 0.0   
2               0.0  ...    0.000000                 0.0   
3               0.0  ...    0.058824                 0.0   
4               0.0  ...    0.500000                 0.0   

   Scandinavian Restaurant  Seafood Restaurant  \
0                 0.034483            0.034483   
1                 0.041667            0.041667   
2                 0.000000            0.000000   
3                 0.000000            0.000000   
4                 0.000000            0.000000   

   Southern / Soul Food Restaurant  Sushi Restaurant  Tapas Restaurant  \
0                         0.068966          0.034483          0.034483   
1                         0.041667          0.083333          0.041667   
2                         0.000000          0.000000          0.000000   
3                         0.058824          0.000000          0.058824   
4                         0.000000          0.000000          0.000000   

   Thai Restaurant  Vegetarian / Vegan Restaurant  Vietnamese Restaurant  
0         0.137931                       0.034483               0.034483  
1         0.125000                       0.041667               0.041667  
2         0.000000                       0.086957               0.217391  
3         0.058824                       0.000000               0.058824  
4         0.000000                       0.000000               0.000000  

[5 rows x 44 columns]

In [29]:
num_top_venues = 5

for hood in pdx_grouped['Neighborhoods']:
    print(hood)
    temp = pdx_grouped[pdx_grouped['Neighborhoods'] == hood].T.reset_index()
    temp.columns = ['venue','freq']
    temp = temp.iloc[1:]
    temp['freq'] = temp['freq'].astype(float)
    temp = temp.round({'freq': 2})
    print(temp.sort_values('freq', ascending=False).reset_index(drop=True).head(num_top_venues))
    print('\n')

Alameda
                             venue  freq
0                  Thai Restaurant  0.14
1               Mexican Restaurant  0.10
2  Southern / Soul Food Restaurant  0.07
3          New American Restaurant  0.07
4        Middle Eastern Restaurant  0.07


Albina
                     venue  freq
0          Thai Restaurant  0.12
1         Sushi Restaurant  0.08
2  New American Restaurant  0.08
3      American Restaurant  0.04
4        Indian Restaurant  0.04


Arleta
                           venue  freq
0          Vietnamese Restaurant  0.22
1             Chinese Restaurant  0.09
2  Vegetarian / Vegan Restaurant  0.09
3             Mexican Restaurant  0.09
4              Korean Restaurant  0.09


Barnes Heights
                venue  freq
0  Italian Restaurant  0.18
1  Mexican Restaurant  0.12
2   French Restaurant  0.06
3     Thai Restaurant  0.06
4    Tapas Restaurant  0.06


Bonneville
                           venue  freq
0            American Restaurant   0.5
1                   

In [30]:
def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    
    return row_categories_sorted.index.values[0:num_top_venues]

In [31]:
num_top_venues = 5

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Neighborhoods']
for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
pdx_ven_sorted = pd.DataFrame(columns=columns)
pdx_ven_sorted['Neighborhoods'] = pdx_grouped['Neighborhoods']

for ind in np.arange(pdx_grouped.shape[0]):
    pdx_ven_sorted .iloc[ind, 1:] = return_most_common_venues(pdx_grouped.iloc[ind, :], num_top_venues)

pdx_ven_sorted

Neighborhoods  1st Most Common Venue 2nd Most Common Venue  \
0              Alameda        Thai Restaurant    Mexican Restaurant   
1               Albina        Thai Restaurant      Sushi Restaurant   
2               Arleta  Vietnamese Restaurant    Mexican Restaurant   
3       Barnes Heights     Italian Restaurant    Mexican Restaurant   
4           Bonneville    American Restaurant            Restaurant   
..                 ...                    ...                   ...   
56  West Portland Park     Mexican Restaurant   American Restaurant   
57        Westmoreland     Italian Restaurant      Sushi Restaurant   
58      Whitwood Court     Mexican Restaurant       Thai Restaurant   
59  Willamette Heights     Italian Restaurant            Restaurant   
60           Woodstock  Vietnamese Restaurant    Mexican Restaurant   

            3rd Most Common Venue            4th Most Common Venue  \
0       Middle Eastern Restaurant  Southern / Soul Food Restaurant   
1         New American Restaurant            Vietnamese Restaurant   
2              Chinese Restaurant    Vegetarian / Vegan Restaurant   
3                      Restaurant                French Restaurant   
4             Dumpling Restaurant                Indian Restaurant   
..                            ...                              ...   
56               Sushi Restaurant               Seafood Restaurant   
57             Seafood Restaurant            Vietnamese Restaurant   
58             Italian Restaurant               Chinese Restaurant   
59              French Restaurant                Korean Restaurant   
60  Vegetarian / Vegan Restaurant              American Restaurant   

        5th Most Common Venue  
0           French Restaurant  
1   Middle Eastern Restaurant  
2           Korean Restaurant  
3         Hawaiian Restaurant  
4         Hawaiian Restaurant  
..                        ...  
56          German Restaurant  
57        American Restaurant  
58         Falafel Restaurant  
59   Mediterranean Restaurant  
60           Sushi Restaurant  

[61 rows x 6 columns]

<h4> As mentioned in the Business Problem section of the notebook. We are trying to help someone intrerested 
to open a business in Oregon, determine which neighborhood would be best. They have an idea for a Fusion Restaurant
in which they wish to combine the spices and flavours of Thai cuisine with the dishes loved in American cuisine.</h4>

<h4> So lets create a new dataframe from the dataframe with top most common venues by selecting those neighborhoods in which Thai cuisine is much loved and features in the 1st Most Common Venue </h4>

In [73]:
df_bestthai = pdx_ven_sorted [pdx_ven_sorted ['1st Most Common Venue'].astype(str).str.contains('Thai Restaurant')]

df_bestthai

Neighborhoods 1st Most Common Venue      2nd Most Common Venue  \
0           Alameda       Thai Restaurant         Mexican Restaurant   
1            Albina       Thai Restaurant           Sushi Restaurant   
25        Irvington       Thai Restaurant  Middle Eastern Restaurant   
28           Kenton       Thai Restaurant         Mexican Restaurant   
30      Laurelhurst       Thai Restaurant         Italian Restaurant   
47      Saint Johns       Thai Restaurant         Mexican Restaurant   
52  University Park       Thai Restaurant         Mexican Restaurant   

              3rd Most Common Venue            4th Most Common Venue  \
0         Middle Eastern Restaurant  Southern / Soul Food Restaurant   
1           New American Restaurant            Vietnamese Restaurant   
25  Southern / Soul Food Restaurant                French Restaurant   
28              American Restaurant          New American Restaurant   
30    Vegetarian / Vegan Restaurant       Modern European Restaurant   
47               Italian Restaurant    Vegetarian / Vegan Restaurant   
52               Seafood Restaurant               Italian Restaurant   

            5th Most Common Venue  
0               French Restaurant  
1       Middle Eastern Restaurant  
25        New American Restaurant  
28       Mediterranean Restaurant  
30               Cuban Restaurant  
47            Japanese Restaurant  
52  Vegetarian / Vegan Restaurant


<h4> Those neighborhoods in which American Cuisine also features in the Most common venues is also of
interest to us as the fusion restaurant will also attract customers who relish dishes of American Cuisine</h4>
<h4> So let us create a new and final dataframe with the Neighborhoods which satisfy both the criterias and have both Thai as well as American Cuisines in the most common venues. The resulting dataframe  will give us the best Neighborhoods for this Fusion Restaurant that our client wants to invest in. </h4>

In [76]:
df_2 = df_bestthai [df_bestthai ['2nd Most Common Venue'].astype(str).str.contains('American Restaurant')]
df_3 = df_bestthai [df_bestthai ['3rd Most Common Venue'].astype(str).str.contains('American Restaurant')]
df_4 = df_bestthai [df_bestthai ['4th Most Common Venue'].astype(str).str.contains('American Restaurant')]
df_5 = df_bestthai [df_bestthai ['5th Most Common Venue'].astype(str).str.contains('American Restaurant')]
df_final = pd.concat([df_2, df_3, df_4, df_5])
df_final

Neighborhoods 1st Most Common Venue      2nd Most Common Venue  \
1         Albina       Thai Restaurant           Sushi Restaurant   
28        Kenton       Thai Restaurant         Mexican Restaurant   
28        Kenton       Thai Restaurant         Mexican Restaurant   
25     Irvington       Thai Restaurant  Middle Eastern Restaurant   

              3rd Most Common Venue    4th Most Common Venue  \
1           New American Restaurant    Vietnamese Restaurant   
28              American Restaurant  New American Restaurant   
28              American Restaurant  New American Restaurant   
25  Southern / Soul Food Restaurant        French Restaurant   

        5th Most Common Venue  
1   Middle Eastern Restaurant  
28   Mediterranean Restaurant  
28   Mediterranean Restaurant  
25    New American Restaurant

<h1> RESULTS </h1>

In [81]:
print ('The best neighborhoods for Thai/American fusion restaurants are :')
print(np.unique(df_final['Neighborhoods']))

The best neighborhoods for Thai/American fusion restaurants are :
['Albina' 'Irvington' 'Kenton']


<h4>By this analysis we got an idea about which restaurants are most liked in Portland, the county seat of the 
Multnomah County. </h4>
<h4> We were also able to determine by this analysis that Mexican Restaurants top the list in most number
of restaurants in the Multnomah County. This can be of interest to someone who is interested to open a
Mexican Restaurant or give a business idea to someone who wants to open a restaurant in Multnomah County
but wants a good analysis of which restaurants are most liked. </h4>

<h1> DISCUSSIONS </h1>

<h4> We carried out this analysis only on the basis of data gathered from Foursquare API. We have not 
considered other factors like demographics, foot traffic, property rates, restaurant prices, if the neighborhood has predominantly businesses 
or residential complexes, schools, proximity to public transportation system etc. Analysis taking all these factors into consideration will yield 
a better result of the best neighborhood. </h4>
<h4> This analysis is a very basic one but it gives us a good idea of what the businesses are operating in Portland,
what kind of food people prefer and which would be good neighborhoods to start a restaurants depending on its Cuisine.</h4>

<h1> CONCLUSION </h1>

<h4> Analysis was done on data to determine the best neighborhoods for a 
Fusion Thai-American Restaurant in Oregon, USA. This analysis involved data collection by web scraping ,
Geopy, Foursquare API. Pandas dataframes were used for the analysis. The result of the analysis gave us 3
good candidate neighborhoods for the Restaurant. We were also able to determine the most loved cuisine.
The analysis can be improved and made more accurate by considereing more parameters rather than just 
venue data.</h4>

<h4> However, in conclusion our business problem was solved. We were able to determine the best neighborhood.
This would be a good trial run for our customer who wants to test the waters in a city not as large,
costly, populated as San Francisco but comes close with regards to demographics and has potential
for tremondous growth. </h4>
